File: PrimeiraAplicacao.Funcional/Funcoes.Funcional/FuncoesAninhadas.fs
module FuncoesAninhadas

let escreveSeNumeroEParOuImpar numero =
    let verificaSeONumeroEPar = numero % 2 = 0 
    let escreveSeNumeroPar() =
        printfn "O número %i é par" numero
    let escreveNumeroImpar() =
        printfn "O número %i é impar" numero
    
    if verificaSeONumeroEPar then
        escreveSeNumeroPar()
    else
        escreveNumeroImpar()

File: PrimeiraAplicacao.Funcional/Funcoes/FuncoesAninhadas.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Funcoes
{
    public class FuncoesAninhadas
    {
        public void EscreveSeNumeroEParOuImpar(int numero)
        {
            bool verificarSeNumeroEPar() => numero % 2 == 0;

            Action escreveNumeroPar = () =>
                Console.WriteLine(string.Format("O número {0} é par", numero));

            Action escreveNumeroImpar = () =>
                Console.WriteLine(string.Format("O número {0} é impar", numero));

            if (verificarSeNumeroEPar())
                escreveNumeroPar();
            else
                escreveNumeroImpar();
        }
    }
}
